deltacat/autoscaler/events/workflow.py

```diff
@@ -0,0 +1,142 @@
+import logging
+from uuid import uuid4
+
+import time
+from typing import Dict, Callable, Union, Tuple, Optional, List, Any
+
+from botocore.exceptions import BotoCoreError
+from ray.autoscaler._private.aws.events import AwsEventManagerBase
+from ray.autoscaler._private.event_system import States
+
+from deltacat import logs
+from deltacat.autoscaler.events.dispatcher import EventDispatcher
+from deltacat.autoscaler.events.event_store import EventStoreClient
+from deltacat.autoscaler.events.exceptions import EventNotFoundException
+
+logging.basicConfig(level=logging.INFO)
+logger = logs.configure_deltacat_logger(logging.getLogger(__name__))
+
+QUERY_EVENTS_MAX_RETRY_COUNTER = 100
+SLEEP_PERIOD_SECONDS = 10
+
+
+def poll_events(events_manager: AwsEventManagerBase,
```

Comment:
Can we separate the AWS Event Manager impl out from a version of this that works for any event manager? As part of this, we may also want to consider (1) putting this inside of a class that provides an expected set of state transitions per implementing job type (and making the state transition map a member of that class) and (2) renaming this from "workflow" to something more specific like "event processor" or "event poller".